pyomo/solvers/plugins/solvers/xpress_direct.py

```diff
@@ -203,11 +205,23 @@ def _apply_solver(self):
                 self._solver_model.setControl(key, contr_type(option))
 
         start_time = time.time()
-        self._solver_model.solve()
+        if self._tee:
+            self._solver_model.solve()
+        else:
+            # In xpress versions greater than or equal 36,
+            # it seems difficult to completely suppress console
+            # output without disabling logging altogether.
+            # As a work around, we capature all screen output
+            # when tee is False.
+            try:
+                with capture_output() as OUT:
+                    self._solver_model.solve()
```

Comment:
Should this be sent to the `self._log_file`?  Also, why is this in a try-except block?

Reply:
The output will be sent to `self._log_file` via the method call on line 171: `self._solver_model.setlogfile(self._log_file)`.

As for the try/except, I wanted to be sure if the `solve` method raised a python Exception that it would be printed to the screen. Perhaps `capture_output()` handles this already, or this shouldn't be the behavior expected from a Pyomo solver with `tee=False`.

Reply:
If the solver raises an exception, the error is printed to the screen at the very top of the call stack (that is, the exception passes out of the `capture_output` triggering the context manager's `__exit__`, so the output will be restored before the exception is ever printed to the scrieen.